### PR TITLE
refactor: apply `webpack-defaults`

### PR DESCRIPTION
**Pull request by d3viant0ne**
Intended to be merged & released as a part of `1.0.0` on a beta `dist-tag` once this has been finished and properly tested.

BREAKING CHANGE: 

Enforces NodeJS > 4.8 via engines 

Closes #25

### CODE REVIEW

File: tests/test-utils.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import path from 'path';
```
Comment (michael-ciniawsky) [inline tests/test-utils.js]: Maybe we should leave this out for now until there is consensus about the initial test setup

File: tests/test-utils.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import webpack from 'webpack';
+import Promise from 'bluebird';
```
Comment (michael-ciniawsky) [inline tests/test-utils.js]: Please no extra lib for `{Promise}` as every supported target has native `{Promise}` support 
`promisify()` can easily be polyfilled by a custom helper or a lightweight lib || `utils.promisify()`

File: tests/test-utils.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  const webpackConfig = getExampleConfig(exampleName);
+  const compiler = webpack(webpackConfig);
+
+  compiler.outputFileSystem = fs;
+
+  const run = Promise.promisify(compiler.run, { context: compiler });
+  const stats = await run();
+
+  const { compilation } = stats;
+  const { errors, warnings, assets, entrypoints, chunks, modules } = compilation;
```
Comment (michael-ciniawsky) [inline tests/test-utils.js]: Destructuring here and returning an {Object} on L69 doesn't really make sense, its better to return the whole `compilation` and pass it to helpers  in the particular test

**example.test.js**
```js
import webpack from 'helpers/compiler'
import { loader, errors, warnings, assets } from 'helpers/utils'  // maybe test-utils/stats-utils

test('Loader', () => {
   const config = { loader: {...options} }

   webpack('path/to/fixture.ext', config)
    .then((compilation) => loader(compilation))
    .then({ result, map, meta } => {
       expect(result).toMatchSnapshot()
       if (map) expect(map).toMatchSnapshot()
       ...
    })
   .catch((err) => err)
})
```

Comment (michael-ciniawsky) [reply tests/test-utils.js]: But thats discussion for [webpack-defaults #68](https://github.com/webpack-contrib/webpack-defaults/issues/68) to not derail the PR here further :)

File: tests/raw-loader.test.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+describe('Raw Loader: Fixture: basic', () => {
+  const fixtureName = 'basic.scss';
+
+  beforeEach(() => {
+
+  });
+
+  test('should load scss file', async () => {
+    const { result } = await runLoadersPromise({ resource: getFixtureResource(fixtureName), loaders });
```
Comment (michael-ciniawsky) [inline tests/raw-loader.test.js]: Maybe just two simple tests (default/escaping(#36)) with `loader.call(ctx, content)` for now as the `raw-loader` doesn't demand any requirements and proper testing in a follow up

File: tests/test-utils.js (hunk 2 above)
Comment (d3viant0ne) [reply tests/test-utils.js]: See comment below & I agree with you entirely. In fact I asked Sean about that shortly after looking at the example. 

`utils.promisify()` iirc isn't available in all our supported Node versions so if we were going to use this setup ( and i'm not ) i was thinking about something like pify.

File: tests/raw-loader.test.js (hunk 4 above)
Comment (d3viant0ne) [reply tests/raw-loader.test.js]: @michael-ciniawsky - Yeah, I was just tinkering with Seans test setup which is complete overkill here but how cumbersome it was in a small repo was what I was curious about.

I'm going to lean this out a bunch more inline with your initial pass

**Pull request closed without merging**